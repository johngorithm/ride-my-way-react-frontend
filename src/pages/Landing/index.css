.container,
#root,
.App {
  height: 100%;
}

#landing-page {
  background-image: url('../../static/images/carpooling-apps-like-lyft-624x351.jpeg');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
}

#landing-page.content .wrapper {
  color: white;
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  right: 0;
  left: 0;
}

#landing-page.content .wrapper p {
  color: white;
  font-size: 20px;
}

#landing-page.content .wrapper h2 {
  color: white;
  font-size: 50px;
  font-family: 'Montserrat', sans-serif;
  margin-bottom: 30px;
}

#landing-page.content .overlay {
  background: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

#landing-page.content .wrapper a .button {
  padding: 10px 40px;
  font-weight: bold;
}
