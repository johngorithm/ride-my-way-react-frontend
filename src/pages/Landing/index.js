import React from 'react';
import { Link } from 'react-router-dom';
import './index.css';

const Landing = () => {
  return (
    <main id="landing-page" className="dropdown content">
      <div className="overlay" />
      <div className="wrapper">
        <h2 className="welcome-title">Welcome</h2>
        <p className="welcome-message">
          Ride My Way is an app that lets you find a ride going your way or
          Create one and get others to their destination
        </p>

        <Link to="/login">
          <button className="button button-blue">GET STARTED</button>
        </Link>
      </div>
    </main>
  );
};

export default Landing;
