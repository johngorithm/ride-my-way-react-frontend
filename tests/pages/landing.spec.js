/* global jest */

import React from 'react';
import { shallow } from 'enzyme';

import  Landing  from 'pages/Landing';


describe('Home Page Tests', () => {
  test('should contain the following elements', () => {
    const wrapper = shallow(<Landing />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('h2').text()).toBe('Welcome');
    expect(wrapper.find('button').exists()).toBe(true)
    expect(wrapper.find('button').text()).toBe('GET STARTED')
    expect(wrapper.find('p').exists()).toBe(true)
  });


  test('should match snapshot', () => {
    const wrapper = shallow(<Landing />);
    expect(wrapper).toMatchSnapshot();
  });

});