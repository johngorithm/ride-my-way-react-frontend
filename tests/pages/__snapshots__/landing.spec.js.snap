// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Tests should match snapshot 1`] = `
<main
  className="dropdown content"
  id="landing-page"
>
  <div
    className="overlay"
  />
  <div
    className="wrapper"
  >
    <h2
      className="welcome-title"
    >
      Welcome
    </h2>
    <p
      className="welcome-message"
    >
      Ride My Way is an app that lets you find a ride going your way or Create one and get others to their destination
    </p>
    <Link
      replace={false}
      to="/login"
    >
      <button
        className="button button-blue"
      >
        GET STARTED
      </button>
    </Link>
  </div>
</main>
`;
